penguin/web/routes.py

```
@@ -5596,27 +5615,60 @@ async def load_context_file(
             "path": str(file_path),
             "filename": file.filename,
-            "content_type": file.content_type,
+            "content_type": content_type,
+            "size_bytes": bytes_written,
         }
+    except HTTPException:
+        if 'file_path' in locals() and file_path.exists():
+            with suppress(FileNotFoundError):
+                file_path.unlink()
+        raise
     except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)